noir-projects/noir-protocol-circuits/crates/types/src/utils/arrays.nr

```diff
@@ -137,6 +137,26 @@ pub fn array_concat<T, let N: u32, let M: u32>(array1: [T; N], array2: [T; M]) -
 }
 
 pub fn array_merge<T, let N: u32>(array1: [T; N], array2: [T; N]) -> [T; N]
```

Comment:
This function seems a little weird in general to me as it assumes that the input arrays cannot be more than half filled, i.e. that `array_length(array1) + array_length(array2) <= N`.

Previously if we had `array_length(array1) + array_length(array2) > N` then the circuit would be unsatisfiable but this new implementation will drop any trailing elements from `array2`. We should add an explicit assertion on the two lengths now.